control-plane-operator/hostedclusterconfigoperator/cmd.go

```diff
@@ -210,6 +215,22 @@ func (o *HostedClusterConfigOperator) Run(ctx context.Context) error {
 	if err := mgr.Add(cpCluster); err != nil {
 		return fmt.Errorf("cannot add CPCluster to manager: %v", err)
 	}
+	var infraConfig *rest.Config
+	var infraCluster cluster.Cluster
+	if _, err := os.Stat(o.InfraKubeconfig); err == nil {
+		infraConfig = operator.CfgFromFile(o.InfraKubeconfig)
+		infraCluster, err = cluster.New(infraConfig, func(opt *cluster.Options) {
+			opt.Namespace = o.Namespace
+			opt.Scheme = api.Scheme
+		})
+		if err != nil {
+			return fmt.Errorf("cannot create infra cluster: %v", err)
+		}
+		if err := mgr.Add(infraCluster); err != nil {
+			return fmt.Errorf("cannot add InfraCluster to manager: %v", err)
+		}
+	}
+
```

Comment:
i've added an else term to default to `cpConfig` and `cpCluster` in case external kubeconfig hasn't been provided.

however, i'm not quite confident what is it going to achieve.
others could call the infra client and get the cp client instead, which could lead to unexpected results.

Reply:
> however, i'm not quite confident what is it going to achieve.
> others could call the infra client and get the cp client instead, which could lead to unexpected results.

what if we added the prefix `Kubevirt` to the InfraConfig and InfraCluster, so it's obvious this only makes sense to use in relation to kubevirt?

Reply:
ok, the `Kubevirt` prefix has been added to these reconciler variables.